### ISSUE TITLE
Enhance weight decay mechanism

### ISSUE THREAD

**Issue by artur-deluca** [artur-deluca/psopt#3]
Instead of providing a tuple as kwarg, the weight (or inertia factor) could be set as a [callback](https://github.com/keras-team/keras/blob/master/keras/callbacks.py) with the following parameters

- starting value
- finishing value (depending on max iteration or other factors)
